### PR TITLE
WebView Blog

### PR DESCRIPTION
**Pull request by ashajgotra**
Android WebViews

### CODE REVIEW

File: _posts/2015-02-14-android-webview.markdown [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+date:   2015-02-14 13:41:00
+category: android
+author: ashajgotra
+---
+
+Android delivers apps in three ways. First is the client-side (native) app developed using the Android SDK and delivered as an installable APK. The second delivery method is for a web application developed using web standards and made available to Android through a native Android web browser application. Finally, we can deliver a web application through an embedded web browser instantiated within a native Android APK using the [WebView](http://developer.android.com/reference/android/webkit/WebView.html) API. In this article, we will explore how to use the WebView API along with strengths and weaknesses compared to native display or browser display.
+
+## WebViews in Android
+
+WebView is the basic building block for displaying web pages in Android. It can be used to display any online content in an Activity or could be used to build your own browser.It provides functionality like zooming, text searches etc. In its basic form it does not enable java script and ignores web errors supporting only basic HTML functionality.
```
Comment (gotoplanb) [inline _posts/2015-02-14-android-webview.markdown]: Is there a more advanced building block, or is this the only way to display a web page for Android?

File: _posts/2015-02-14-android-webview.markdown [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+# WebView Use-Cases
+
+Making use of `Webviews` makes sense:
+
+- when rendering information that requires to be updated regularly e.g. user-guide. In such case it is better to display online document in a webview.
+- if app requires to be online to fetch data e.g. email. Rather than retrieving data on network call and displaying in layout it is better to design web page tailored for Android devices and load the web page in webview.
+
```
Comment (gotoplanb) [inline _posts/2015-02-14-android-webview.markdown]: Add similar list of when WebView does not make sense.

**Comment by gotoplanb** [comment]
@ashajgotra Great start! I left a couple in-line comments primarily to add a short list of scenarios where `WebView` is the wrong choice. 

@danish7 Can you take a look at this post for technical accuracy? If looks good to you, then comment with the Ship-It Squirrel `:shipit:` so that I can give another read-through for grammar fixes.

**Comment by danish7** [comment]
@gotoplanb Sure, i'll take a look.

**Comment by ashajgotra** [comment]
@gotoplanb Will add that!!

File: _posts/2015-02-14-android-webview.markdown [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+# Using JavaScript
+
+`JavaScript` in webviews is disabled by default. It can be enabled as shown below:
+
+	{% highlight java %}
+	
+	WebView myWebView = (WebView) findViewById(R.id.webview);
+	WebSettings webSettings = myWebView.getSettings();
+	webSettings.setJavaScriptEnabled(true);
+
```
Comment (danish7) [inline _posts/2015-02-14-android-webview.markdown]: I think you should highlight how to enable built in zoom settings that you talk about earlier.

**Comment by danish7** [comment]
@ashajgotra Looks good to me. I would also add some bits about supporting different device densities. Read here : http://developer.android.com/reference/android/webkit/WebView.html

File: _posts/2015-02-14-android-webview.markdown (hunk 3 above)
Comment (ashajgotra) [inline _posts/2015-02-14-android-webview.markdown]: Alright!

File: _posts/2015-02-14-android-webview.markdown (hunk 1 above)
Comment (ashajgotra) [inline _posts/2015-02-14-android-webview.markdown]: In Android we can either use the native browser or the WebView to display web content. WebView are much more flexible and offer the added functionality to build our custom browser.

**Comment by danish7** [comment]
:shipit: @gotoplanb

**Comment by tikipatel** [comment]
test comment selfie-0

**Pull request merged**